src/main/java/net/neoforged/neoforge/transfer/resources/ItemResource.java

```diff
@@ -0,0 +1,303 @@
+        ItemStack stack = innerStack.copy();
+        stack.set(type, data);
+        return new ItemResource(stack);
+    }
+
+    @Override
+    public <D> ItemResource with(Supplier<DataComponentType<D>> type, D data) {
+        return with(type.get(), data);
+    }
+
+    @Override
+    public ItemResource without(DataComponentType<?> type) {
+        ItemStack stack = innerStack.copy();
+        stack.remove(type);
+        return new ItemResource(stack);
+    }
+
+    @Override
+    public ItemResource without(Supplier<? extends DataComponentType<?>> type) {
+        return without(type.get());
+    }
+
+    @Override
+    public DataComponentMap getComponents() {
+        return innerStack.immutableComponents();
+    }
+
+    @Override
+    public DataComponentPatch getComponentsPatch() {
+        return innerStack.getComponentsPatch();
+    }
+
+    public ItemStack toStack() {
+        return toStack(1);
+    }
+
+    public ItemStack toStack(int count) {
+        if (count == 0 || this.isEmpty()) return ItemStack.EMPTY;
+        return this.innerStack.copyWithCount(count);
+    }
+
+    public List<ItemStack> toStacks(int count) {
+        if (count == 0 || isEmpty())
+            return Collections.emptyList();
+
+        int maxStackSize = getMaxStackSize();
+        int stackCount = count / maxStackSize;
+        List<ItemStack> stacks = new ArrayList<>(stackCount + 1);
+        for (int i = 0; i < stackCount; i++) {
+            stacks.add(toStack(maxStackSize));
+        }
+        int remainder = count % maxStackSize;
+        if (remainder > 0) {
+            stacks.add(toStack(remainder));
+        }
+        return stacks;
+    }
+
+    public int getMaxStackSize() {
+        return innerStack.getMaxStackSize();
+    }
+
+    /**
+     * @return The hover name of the {@link ItemStack}
+     */
+    public Component getHoverName() {
+        return innerStack.getHoverName();
+    }
+
+    public boolean canEquip(EquipmentSlot slot, LivingEntity entity) {
+        return innerStack.canEquip(slot, entity);
+    }
+
+    public boolean canUnequip() {
+        return !EnchantmentHelper.has(innerStack, EnchantmentEffectComponents.PREVENT_ARMOR_CHANGE);
+    }
+
+    public ResourceStack<ItemResource> withAmount(int amount) {
+        if (amount == 0 || isEmpty()) return ItemResource.EMPTY_STACK;
+        return ResourceStack.of(this, amount, ItemResource.EMPTY_STACK);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        return obj instanceof ItemResource other && other.is(innerStack);
+    }
+
+    @Override
+    public int hashCode() {
+        return ItemStack.hashItemAndComponents(innerStack);
+    }
+
+    @Override
+    public String toString() {
+        return innerStack.getItem().toString();
+    }
+
+    /**
+     * @return the full value and data components in string form
+     */
+    public String toExpandedString() {
```

Comment:
Why is this necessary? What uses this function?

Surely any caller wanting this information can do this themselves.